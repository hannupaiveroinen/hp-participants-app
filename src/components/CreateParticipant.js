import React, { Component } from 'react';
import { addParticipant } from '../redux/actions';
import { connect } from "react-redux";
import ReactFormInputValidation from "react-form-input-validation";

class CreateParticipant extends Component {
    constructor(props) {
        super(props);
        this.state = {
            fields: {
                name: '',
                email: '',
                phone: ''
            },
            errors: {}
        }

        this.form = new ReactFormInputValidation(this);
        this.form.useRules({
            name: "required",
            email: "required|email",
            phone: "required|numeric|digits_between:10,12",
        });

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit = () => {
        this.props.addParticipant(this.state.fields);
        this.setState({
            fields: {
                name: '',
                email: '',
                phone: ''
            }
        });
    }

    render() {
        return (
            <div className='create-participant-form'>
                <h1>List of participants</h1>
                <div>
                    <input type="text"
                        name="name"
                        value={this.state.fields.name}
                        placeholder="Full name"
                        onBlur={this.form.handleBlurEvent}
                        onChange={this.form.handleChangeEvent}
                        style={{ width: 180 }} />
                    <input type="email"
                        name="email"
                        value={this.state.fields.email}
                        placeholder="E-mail address"
                        onBlur={this.form.handleBlurEvent}
                        onChange={this.form.handleChangeEvent}
                        style={{ width: 320 }} />
                    <input type="text"
                        name="phone"
                        value={this.state.fields.phone}
                        placeholder="Phone number"
                        onBlur={this.form.handleBlurEvent}
                        onChange={this.form.handleChangeEvent}
                        style={{ width: 178 }} />
                    <button disabled={this.state.errors.name || this.state.errors.email || this.state.errors.phone} onClick={this.handleSubmit}>Add new</button>
                </div>
                <div style={{ padding: 0 }} >
                    <label style={{ width: 200 }} className="error">
                        {this.state.errors.name ? this.state.errors.name : ""}
                    </label>
                    <label style={{ width: 340 }} className="error">
                        {this.state.errors.email ? this.state.errors.email : ""}
                    </label>
                    <label style={{ width: 200 }} className="error">
                        {this.state.errors.phone ? this.state.errors.phone : ""}
                    </label>
                </div>
            </div>
        );
    }
}

export default connect(
    null,
    { addParticipant }
)(CreateParticipant);
