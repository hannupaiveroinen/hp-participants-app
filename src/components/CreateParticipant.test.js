import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CreateParticipant from './CreateParticipant';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);

configure({ adapter: new Adapter() });

describe("CreateParticipant", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <CreateParticipant />
      </Provider>
    );
  });

  it("should render my component", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
   
  it('should dispatch an action on add participant click', () => {
    // TODO
  });
});