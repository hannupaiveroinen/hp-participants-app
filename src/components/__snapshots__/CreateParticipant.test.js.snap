exports[`CreateParticipant should render my component 1`] = `
<div
  className="create-participant-form">
  <h1>
    List of participants
  </h1>
  <div>
    <input
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Full name"
      style={
        Object {
          "width": 180,
        }
      }
      type="text"
      value="" />
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="E-mail address"
      style={
        Object {
          "width": 320,
        }
      }
      type="email"
      value="" />
    <input
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Phone number"
      style={
        Object {
          "width": 178,
        }
      }
      type="text"
      value="" />
    <button
      disabled={undefined}
      onClick={[Function]}>
      Add new
    </button>
  </div>
  <div
    style={
      Object {
        "padding": 0,
      }
    }>
    <label
      className="error"
      style={
        Object {
          "width": 180,
        }
      }>
      
    </label>
    <label
      className="error"
      style={
        Object {
          "width": 320,
        }
      }>
      
    </label>
    <label
      className="error"
      style={
        Object {
          "width": 180,
        }
      }>
      
    </label>
  </div>
</div>
`;
